### PR TITLE
Center type and method importation around Cecil

### LINKED ISSUE #161: Method lookup: get rid of reflection code
As of now, `Cesium.CodeGen.Extensions.TypeSystemEx.MethodLookup` does lookup on referenced assemblies using the runtime reflection. This means that to look up a method, Cesium has to load the referenced assembly.

This is not right, since not every assembly may even be loaded by the compiler (due to architecture or platform differences between the compiler runtime and the application target runtime).

We should inspect the referenced assemblies using Cecil instead of the runtime reflection

See also: `// TODO[#161]` in the compiler code.

### PR DESCRIPTION
**Pull request by kant2002**
Reduce usage of `typeof` and Reflection across codebase
I did not change all usages, since some of them require factoring code differently, and it produce much more noise, and would be hard to review

**Notes from F.:**

Closes #161.

### CODE REVIEW

**Review by ForNeVeR — CHANGES_REQUESTED** [review]
Overall looks good, but there's one small issue.

File: Cesium.CodeGen/Ir/Expressions/SubscriptingExpression.cs [hunk 1]
```diff
@@ -32,9 +32,13 @@ public IExpression Lower()
...
     {
-        if (_expression is not IdentifierExpression identifier)
-            throw new NotImplementedException("Subscription supported only for IdentifierConstantExpression");
-
-        return new LValueArrayElement(identifier.Resolve(scope), _index);
+        switch (_expression) {
+            case IdentifierExpression identifier:
+                return new LValueArrayElement(identifier.Resolve(scope), _index);
+            case PointerMemberAccessExpression pointerMemberAccess:
+                return new LValueArrayElement(pointerMemberAccess.Resolve(scope), _index);
```
[the remark this answers is not in the document]
Comment (kant2002) [reply Cesium.CodeGen/Ir/Expressions/SubscriptingExpression.cs]: Indeed it was leftover

**Comment by kant2002** [comment]
I think I'm done here too.

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**